pkg/operator/operator.go

```diff
@@ -282,3 +287,21 @@ func WaitForCRDs(ctx context.Context, timeout time.Duration, config *rest.Config
 	log.Info("all required CRDs are available")
 	return nil
 }
+
+// ensureToken ensures we can get a token for the Azure environment. Note that this doesn't actually
+// use the token for anything, it just checks that we can get one.
+func ensureToken(cfg *auth.Config) error {
+	creds, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return err
+	}
+	cloudEnv := azclient.EnvironmentFromName(cfg.Cloud)
+	_, err = creds.GetToken(context.Background(), policy.TokenRequestOptions{
```

Comment:
recommend to set a context with short timeout as the GetToken call could block for unknown reason